Engine/DataStream.cs

```diff
@@ -52,14 +52,13 @@ public DataStream(IDataFeed feed)
         {
             do
             {
-                TimeSlice timeSlice;
-                while (_feed.Bridge.TryDequeue(out timeSlice))
+                foreach (var timeSlice in _feed.Bridge.GetConsumingEnumerable())
                 {
                     AlgorithmTime = timeSlice.Time;
                     yield return timeSlice.Data;
                 }
-            }
-            while (!_feed.LoadingComplete && !_feed.Bridge.IsEmpty);
+            } 
+            while (!_feed.Bridge.IsCompleted);
```

Comment:
do we still technically need this do/while loop here? My understanding is the GetConsumingEnumerable will never finish until IsCompleted=true